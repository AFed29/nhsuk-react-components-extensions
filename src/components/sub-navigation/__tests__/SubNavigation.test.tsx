import { shallow } from 'enzyme';
import React from 'react';
import SubNavigation from '../SubNavigation';

describe('SubNavigation', () => {
  it('matches snapshot', () => {
    const component = shallow(<SubNavigation />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should have correct classes applied', () => {
    const component = shallow(<SubNavigation />);
    const unorderedList = component.find('.nhsuk-sub-navigation__list');

    expect(component.hasClass('nhsuk-sub-navigation')).toBeTruthy();
    expect(unorderedList.exists()).toBeTruthy();

    component.unmount();
  });

  describe('SubNavigationItem', () => {
    it('matches snapshot', () => {
      const component = shallow(<SubNavigation.Item href="#">Hello world</SubNavigation.Item>);
      expect(component).toMatchSnapshot();
      component.unmount();
    });

    it('should have the correct classes applied', () => {
      const component = shallow(<SubNavigation.Item href="#">Hello world</SubNavigation.Item>);

      expect(component.hasClass('nhsuk-sub-navigation__item')).toBeTruthy();

      component.unmount();
    });

    it('passes inner text correctly', () => {
      const component = shallow(
        <SubNavigation.Item href="https://nhs.uk">Hello world</SubNavigation.Item>,
      );
      expect(component.text()).toBe('Hello world');
      component.unmount();
    });
  });
});
