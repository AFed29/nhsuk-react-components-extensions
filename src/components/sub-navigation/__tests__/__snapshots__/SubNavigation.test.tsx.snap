// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNavigation SubNavigationItem matches snapshot 1`] = `
<li
  className="nhsuk-sub-navigation__item"
>
  <a
    className="nhsuk-sub-navigation__link"
    href="#"
  >
    Hello world
  </a>
</li>
`;

exports[`SubNavigation matches snapshot 1`] = `
<nav
  className="nhsuk-sub-navigation"
>
  <ul
    className="nhsuk-sub-navigation__list"
  />
</nav>
`;
